src/v/pandaproxy/rest/handlers.cc

```diff
@@ -181,13 +193,14 @@ get_topics_records(server::request_t rq, server::reply_t rp) {
       max_bytes);
 
     return rq.service()
-      .client()
-      .local()
-      .fetch_partition(std::move(tp), offset, max_bytes, timeout)
+      .client_cache()
+      .invoke_on_cache(rq.user, handler)
       .then([res_fmt, rp = std::move(rp)](kafka::fetch_response res) mutable {
           ::json::StringBuffer str_buf;
           ::json::Writer<::json::StringBuffer> w(str_buf);
 
+          // This serialization causes a shard missmatch
+          // _owner_shard == ss::this_shard_id()' iobuf.h:368 - Shard missmatch
           ppj::rjson_serialize_fmt(res_fmt)(w, std::move(res));
```

Comment:
@BenPope I'm getting a crash here due to shard missmatch. I see you have comments about changing this serialize but what are those alternatives?